### PR TITLE
Updating core APIs for uint64 Request Priority

### PR DESCRIPTION
**Pull request by nnshah1**
Updating request priority apis to support uint64.

See discussion here https://github.com/triton-inference-server/common/pull/82#issuecomment-1547650136.

Major changes:

- Add TRITONSERVER_InferenceRequestPriorityUint64
- Add TRITONSERVER_InferenceRequestSetPriorityUint64
- Update handling of uint64 fields in model config to json
- Add error handling for overflow

### CODE REVIEW

File: include/triton/core/tritonserver.h [hunk 1]
```diff
@@ -1078,6 +1092,17 @@ TRITONSERVER_DECLSPEC struct TRITONSERVER_Error* TRITONSERVER_InferenceRequestPr
 TRITONSERVER_DECLSPEC struct TRITONSERVER_Error*
```
Comment (rmccorm4) [inline include/triton/core/tritonserver.h:1092]: Should SetPriority get similar deprecation comment?
```
/// Deprecated. See TRITONSERVER_InferenceRequestSetPriorityU64 instead.
```

File: src/tritonserver.cc [hunk 2]
```diff
@@ -1692,6 +1692,25 @@ TRITONSERVER_InferenceRequestSetCorrelationIdString(
...
+  if (error != nullptr) {
+    return error;
+  }
+  if (temp <= std::numeric_limits<uint32_t>::max()) {
+    *priority = temp;
+    return nullptr; // Success
+  }
+  return TRITONSERVER_ErrorNew(TRITONSERVER_ERROR_INVALID_ARG,
+			       (std::string(
+				   "request priority overflows uint32_t, priority=") +
```
Comment (Tabrizian) [inline src/tritonserver.cc:1707]: Perhaps updating to error message to tell the user to use "TRITONSERVER_InferenceRequestPriorityU64" instead.

File: include/triton/core/tritonserver.h [hunk 3]
```diff
@@ -1068,6 +1070,18 @@ TRITONSERVER_InferenceRequestSetCorrelationIdString(
...
     struct TRITONSERVER_InferenceRequest* inference_request, uint32_t* priority);
 
+/// Get the priority for a request. The default is 0 indicating that
+/// the request does not specify a priority and so will use the
+/// model's default priority.
+///
+/// \param inference_request The request object.
+/// \param priority Returns the priority level.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC struct TRITONSERVER_Error* TRITONSERVER_InferenceRequestPriorityU64(
```
[the remark this answers is not in the document]
Comment (nnshah1) [reply include/triton/core/tritonserver.h:1080]: would rather not break up 'Request' and 'Priority', could be:
`TRITONSERVER_InferenceRequestPriorityUInt64()`

File: src/tritonserver.cc (hunk 2 above)
Comment (nnshah1) [reply src/tritonserver.cc:1707]: will update when name is settled

File: include/triton/core/tritonserver.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tabrizian) [reply include/triton/core/tritonserver.h:1080]: Looks good!

[the remark this answers is not in the document]
Comment (rmccorm4) [reply include/triton/core/tritonserver.h]: I'm happy with `TRITONSERVER_InferenceRequestPriorityUInt64()`

File: include/triton/core/tritonserver.h [hunk 4]
```diff
@@ -1078,6 +1092,17 @@ TRITONSERVER_DECLSPEC struct TRITONSERVER_Error* TRITONSERVER_InferenceRequestPr
...
+  
+/// Set the priority for a request. The default is 0 indicating that
+/// the request does not specify a priority and so will use the
+/// model's default priority.
+///
+/// \param inference_request The request object.
+/// \param priority The priority level.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC struct TRITONSERVER_Error*
+TRITONSERVER_InferenceRequestSetPriorityU64(
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply include/triton/core/tritonserver.h]: I'm happy with `TRITONSERVER_InferenceRequestPriorityUInt64()` as mentioned in other [thread](https://github.com/triton-inference-server/core/pull/214#discussion_r1221892581)

**Review by Tabrizian — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]

**Pull request merged**